src/utils/scraping.py

```diff
@@ -1,74 +1,101 @@
         json.dump(data, f)
-    return data
\ No newline at end of file
+
+    return data
+
+
+def _create_intel_database():
+    """
+    Create the database of Intel CPUs and their TDPs, scraping information from the intel website.
+
+    """
+    def _isintel(tag):
+        return tag.has_attr('data-value') and 'Intel' in tag['data-value']
+    
+    # initialize the dictionary
+    intel_database = dict()
+
+    # get the list of processors families from the intel website
+    search_url = "https://www.intel.com/content/www/us/en/products/details/processors.html"
+    soup = BeautifulSoup(requests.get(search_url).content, 'html.parser')
+    # find the title class
+    group_list = soup.find_all("div", attrs={"class": "group-title has-name"})
+    # iterate over processor groups
+    for group in group_list:
+        # get href, insert 'products' before .html and transform href to url
+        group_url = 'https://www.intel.com' + group.find("a")['href'].replace('.html', '/products.html')
+        soup = BeautifulSoup(requests.get(group_url).content, 'html.parser')
+        # stop the loop for a random time between 0 and 1 seconds to avoid being blocked
+        time.sleep(np.random.rand())
+        # isolate the list of processors by looking for the data-value that contains 'Intel'
+        processor_list = soup.find_all(_isintel)
+        # if it is empty, skip
+        if not processor_list:
+            continue
+        # iterate over processors
+        for processor in processor_list:
+            # get href and transform to url
+            processor_url = 'https://www.intel.com' +  processor.find("a")['href']
+            # request the processor page
+            soup = BeautifulSoup(requests.get(processor_url).content, 'html.parser')
+            # stop the loop for a random time between 0 and 1 seconds to avoid being blocked
+            time.sleep(np.random.rand())
+            # get the tdp
+            processor_specs = soup.find_all("div", attrs={"class": "row tech-section-row"})
+            # iterate over specs
+            for spec in processor_specs:
+                # check if tech-label contains 'TDP'
+                if spec.find("div", attrs={"class": "tech-label"}) is not None:
+                    if 'TDP' in spec.find("div", attrs={"class": "tech-label"}).text:
+                        # get the tdp
+                        tdp = spec.find("div", attrs={"class": "tech-data"}).text
+                        # retain only the numbers including the decimal points, if present
+                        tdp = float(re.findall(r'\d+\.\d+|\d+', tdp)[0])
+                        # store in dictionary
+                        intel_database[processor['data-value']] = tdp
+                        # break the loop over specs, to speed up the process
+                        break
+    # save the dictionary in a json file
+    # get the path to the data folder
+    path = [x[0] for x in os.walk('..') if 'data' in x[0]][0]
```

Comment:
Is there a reason why here you keep the old method of getting the path?

Reply:
nope, indeed i should update also this line. thanks